src/header/internals/item.rs

```diff
@@ -82,6 +82,14 @@ impl Item {
         }
         self.typed.get_mut(tid).map(|typed| unsafe { typed.downcast_mut_unchecked() })
     }
+
+    pub fn into_typed<H: Header>(self) -> Option<H> {
```

Comment:
Oh right. We can convert the `*mut T` into a `Box<T>` for free, with `Box::from_raw`, and then we can deref move out of the box:

```rust
*Box::from_raw(mem::transmute::<*mut _, (*mut (), *mut ())>(Box::into_raw(self)).0 as *mut T)
```